rust-runtime/aws-smithy-observability/src/lib.rs

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Automatically managed default lints */
+#![cfg_attr(docsrs, feature(doc_auto_cfg))]
+/* End of automatically managed default lints */
+#![warn(
+    missing_docs,
+    rustdoc::missing_crate_level_docs,
+    unreachable_pub,
+    rust_2018_idioms
+)]
+
+//! Smithy Observability
+//TODO(smithyobservability): once we have finalized everything and integrated metrics with our runtime
+// libraries update this with detailed usage docs and examples
+
+pub mod attributes;
```

Comment:
Several of these modules _probably_ should be private and just re-export their type instead.

```rs
use aws_smithy_observability::Error;
use aws_smithy_observability::Attributes`
```

vs

```rs
use aws_smithy_observability::error::Error;
use aws_smithy_observability::attributes::Attributes`
```

In other words consider how you'd like the API to be exposed and consumed.

Also missing docs on public modules.

Reply:
Added some `pub use`s for these types in https://github.com/smithy-lang/smithy-rs/pull/3912/commits/0559e8b54fb41bcf04125e2e3cedeec87b24a9f9

The public modules are already documented, but the docs are in the module file instead of in `lib.rs`. I prefer keeping them there because I think it is cleaner than putting them all in `lib.rs`, but will move them if that is a consensus.